fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -38,7 +43,13 @@ def __init__(self,
             raise RuntimeError("quality_metric must derive from QualityMetric")
         self.quality_metric = quality_metric
 
-    def fit(self, X, Y, **kwargs):
+    def fit(self, X, y, **kwargs):
+        if X is None:
+            raise ValueError(self._MESSAGE_X_NONE)
+
+        if y is None:
+            raise ValueError(self._MESSAGE_Y_NONE)
+
         if self._KW_AUX_DATA in kwargs:
             aux_data = kwargs[self._KW_AUX_DATA]
```

Comment:
To be honest I doubt anyone is going to read the code to find out that aux_data is a possible kwarg. Why not just make it part of the signature with
```
def fit(self, X, y, aux_data=None, **kwargs)

Reply:
There's also no documentation that says that it's an option to pass it.

Reply:
Have changed `aux_data` to being named. Documentation... waiting on details like this. Also, I understand there are multiple doc standards we could follow?